clang/lib/AST/Availability.cpp

```diff
@@ -16,33 +16,101 @@
 #include "clang/AST/Decl.h"
 #include "clang/Basic/TargetInfo.h"
 
-namespace clang {
+namespace {
+
+struct AvailabilitySet {
+  llvm::SmallVector<clang::AvailabilityInfo> Availabilities;
```

Comment:
Can you mark this private, its not clear otherwise that one should be going through `insert`